packages/kit/src/utils/url.js

```diff
@@ -0,0 +1,35 @@
+const absolute = /^([a-z]+:)?\/?\//;
+
+/**
+ * @param {string} base
+ * @param {string} path
+ */
+export function resolve(base, path) {
+	const base_match = absolute.exec(base);
+	const path_match = absolute.exec(path);
+
+	if (!base_match) {
+		throw new Error(`bad base path: "${base}"`);
+	}
+
+	const baseparts = path_match ? [] : base.slice(base_match[0].length).split('/');
+	const pathparts = path_match ? path.slice(path_match[0].length).split('/') : path.split('/');
+
+	baseparts.pop();
+
+	for (let i = 0; i < pathparts.length; i += 1) {
+		const part = pathparts[i];
+		if (part === '.') continue;
+		else if (part === '..') baseparts.pop();
+		else baseparts.push(part);
+	}
+
+	const prefix = (path_match && path_match[0]) || (base_match && base_match[0]) || '';
+
+	return `${prefix}${baseparts.join('/')}`;
+}
+
+/** @param {string} path */
+export function is_root_relative(path) {
```

Comment:
I wonder if `is_absolute_path` might be a clearer name?

Reply:
my understanding of the terminology is that `https://example.com/foo` is an absolute path, `//example.com/foo` is a protocol-relative path, and `/foo` is a root-relative path